packages/amplify-graphql-schema-generator/src/ts-schema-generator/helpers.ts

```diff
@@ -147,19 +164,61 @@ export const createSchema = (schema: Schema, config?: DataSourceGenerateConfig):
     throw new Error('No valid tables found. Make sure at least one table has a primary key.');
   }
 
+  const nullableEnumFields = schema.getModels().map((model) =>
+    model
+      .getFields()
+      .filter((field) => field.type.kind === 'Enum')
+      .map((field) => {
+        if (field.type.kind === 'Enum') {
+          return createEnums(field.type);
+        } else {
+          return undefined;
+        }
+      }),
+  );
+
+  const requiredEnumFields = schema.getModels().map((model) =>
+    model
+      .getFields()
+      .filter((field) => field.type.kind === 'NonNull' && field.type.type.kind === 'Enum')
+      .map((field) => {
+        if (field.type.kind === 'NonNull' && field.type.type.kind === 'Enum') {
+          return createEnums(field.type.type);
+        } else {
+          return undefined;
+        }
+      }),
```

Comment:
Is checking `field.type.kind` and `field.type.type.kind` redundant because of the filter?
```suggestion
      .map((field) => createEnums(field.type.type))
```